lightkurve/targetpixelfile.py

```diff
@@ -375,10 +380,60 @@ def _parse_aperture_mask(self, aperture_mask):
                 aperture_mask = np.ones((self.shape[1], self.shape[2]), dtype=bool)
             elif aperture_mask == 'pipeline':
                 aperture_mask = self.pipeline_mask
+            elif aperture_mask == 'threshold':
+                if 'threshold' in kwargs:
+                    threshold = kwargs['threshold']
+                else:
+                    threshold = 3.
+                aperture_mask = self._threshold_mask(threshold)
         self._last_aperture_mask = aperture_mask
         return aperture_mask
 
-    def centroids(self, aperture_mask='pipeline'):
+    def _threshold_mask(self, threshold):
+        """Identify region that is above the threshold and contiguous
+        with the central pixel
+
+        Parameters
+        ----------
+        threshold : float
+            A value for the number of sigma above the background to draw the
+            aperture.
+
+        Returns
+        -------
+        aperture_mask : ndarray
+            2D boolean numpy array containing `True` for selected pixels.
+        """
+
+        # calculate a median image
+        median_image = np.nanmedian(self.flux + self.flux_bkg, axis=0)
+        vals = median_image[np.isfinite(median_image)].flatten()
+
+        # calculate what the theshold value is in counts
+        mad_cut = (1.4826 * MAD(vals) * threshold) + np.median(median_image)
```

Comment:
Are you assuming that the background is the median of the image here? That isn't always a strong assumption.